napari/_lazy.py

```diff
@@ -42,6 +42,11 @@ def __getattr__(name):
         # see: https://github.com/napari/napari/issues/1347
         from scipy import stats  # noqa: F401
 
+        # Import added to fix circular imports
+        from .viewer import Viewer
```

Comment:
🤔

very strange!  do you know _why_ this fixes it?

Reply:
@sofroniewn figured this one out.  

All I can say about it is that it before #2816 was merged, the viewer was imported just before this line was run in the napari `__init`` file:
```
_magicgui.register_types_with_magicgui()
```
That's all I got though...

Reply:
ok, lemme see if we can fix this a better way (by moving the magicgui initialization)